gef.py

```diff
@@ -221,59 +222,271 @@ def highlight_text(text):
+        if not is_remote_debug():
+            return f(*args, **kwargs)
+        else:
+            warn("This command cannot work for remote sessions.")
+
+    return wrapper
+
+
+def deprecated(solution: str = "") -> Callable:
+    """Decorator to add a warning when a command is obsolete and will be removed."""
+    def decorator(f: Callable) -> Callable:
+        @functools.wraps(f)
+        def wrapper(*args: Tuple, **kwargs: Dict) -> Any:
+            msg = f"'{f.__name__}' is deprecated and will be removed in a feature release. "
+            if solution:
+                msg += solution
+            warn(msg)
+            return f(*args, **kwargs)
+
+        if not wrapper.__doc__:
+            wrapper.__doc__ = ""
+        wrapper.__doc__ += f"\r\n`{f.__name__}` is **DEPRECATED** and will be removed in the future.\r\n{solution}"
+        return wrapper
+    return decorator
+
+
+def experimental_feature(f: Callable) -> Callable:
+    """Decorator to add a warning when a feature is experimental."""
+
+    @functools.wraps(f)
+    def wrapper(*args: Tuple, **kwargs: Dict) -> Any:
+        warn("This feature is under development, expect bugs and unstability...")
+        return f(*args, **kwargs)
+
+    return wrapper
+
+
+def only_if_gdb_version_higher_than(required_gdb_version) -> Callable:
+    """Decorator to check whether current GDB version requirements."""
+
+    def wrapper(f: Callable) -> Callable:
+        def inner_f(*args: Tuple, **kwargs: Dict) -> None:
+            if GDB_VERSION >= required_gdb_version:
+                f(*args, **kwargs)
+            else:
+                reason = f"GDB >= {required_gdb_version} for this command"
+                raise OSError(reason)
+        return inner_f
+    return wrapper
+
+
+def only_if_current_arch_in(valid_architectures: List) -> Callable:
+    """Decorator to allow commands for only a subset of the architectured supported by GEF.
+    This decorator is to use lightly, as it goes against the purpose of GEF to support all
+    architectures GDB does. However in some cases, it is necessary."""
+
+    def wrapper(f: Callable) -> Callable:
+        def inner_f(*args: Tuple, **kwargs: Dict) -> None:
+            if gef.arch in valid_architectures:
+                f(*args, **kwargs)
+            else:
+                reason = f"This command cannot work for the '{gef.arch.arch}' architecture"
+                raise OSError(reason)
+        return inner_f
+    return wrapper
+
+
+def only_if_events_supported(event_type) -> Callable:
+    """Checks if GDB supports events without crashing."""
+    def wrap(f: Callable) -> Callable:
+        def wrapped_f(*args: Tuple, **kwargs: Dict) -> Any:
+            if getattr(gdb, "events") and getattr(gdb.events, event_type):
+                return f(*args, **kwargs)
+            warn("GDB events cannot be set")
+        return wrapped_f
+    return wrap
+
+
+def FakeExit(*args, **kwargs) -> None:
```

Comment:
```suggestion
def FakeExit(__status: object = ...) -> NoReturn:
```

This change requires to also import `NoReturn` from `typing` as suggested at the beginning of the file.